src/main/java/Duke.java

```diff
@@ -1,10 +1,264 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    int index;
+    List<Task> ls;
+
+    public Duke() {
+        this.index = 0;
+        this.ls = new ArrayList<>();
+    }
+
+    // prints introduction of klaun
+    public void introDuke() {
+        System.out.println("I'm Klaun (=^.^=) How are you doing today ?\n");
+        System.out.println("Is there anything I can help you with ?\n");
+        System.out.println("<------------------------------------------------------------>\n");
+    }
+
+    public int getIndex() {
+        return this.index;
+    }
+
+    // prints output based on command
+    public void sayBye() {
+        System.out.println("++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n");
+        System.out.println("oh man ... bye ~~ o.o \n");
+        System.out.println("++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++\n");
+    }
+
+    // prints list of tasks
+    public void getList() {
+        System.out.println("##############################################################\n");
+        System.out.println("Here's your amazing task list :");
+        for (int i = 0; i < this.index; i++) {
+            Task task = this.ls.get(i);
+            String item = task.getItem();
+            String status = task.getStatus();
+            String taskType = task.getSign();
+            int num = i + 1;
+
+            if (task instanceof Todo) { // if task is to-do
+                System.out.println(num + ". " + taskType + " " + status + " " + item);
+            } else if (task instanceof Deadline) { // if task is deadline
+                Deadline actualTask = (Deadline) task;
+                System.out.println(num + ". " + taskType + " " +  status + " " + item + " -> by :" + actualTask.getDeadline());
+            } else { // if task type if event
+                Event actualTask = (Event) task;
+                System.out.println(num + ". " + taskType + " " + status + " " + item + " -> at :" + actualTask.getTime());
+            }
+
+        }
+
+        System.out.println(" ");
+        System.out.println("##############################################################\n");
+    }
+
+    // adds task to list
+    public void addToList(Task task) {
+        this.ls.add(task);
+        this.index++; // increment index
+
+        // print confirmation of adding task
+        this.confirmAddTask(task);
+    }
+
+    // mark task as done
+    public void markDone(int index) {
+        this.ls.get(index).markAsDone();
+
+        Task task = this.ls.get(index);
+        String item = task.getItem();
+        String status = task.getStatus();
+        String taskType = task.getSign();
+        int num = index + 1;
+
+        System.out.println("==============================================================\n");
+        System.out.println("Yayyyy !! Letsgedditt");
+        if (task instanceof Todo) { // if task is to-do
```

Comment:
adding on to the comment above about adding a toString() method, this could prevent multiple copy pasting of the same chunk of code :-)